walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -121,6 +126,69 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeIsConsideredWithOptionalAllDayAttendee() {
+    // Have each person have different events and an optional attendee with an all-day event.
+    // We should see two options because each person has split the restricted times.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |~~~~~~~~~~~~~~C~~~~~~~~~~~~~~|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_START, TIME_END, true),
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void everyAttendeeIsConsideredWithOptionalEarlyAttendee() {
+    // Have each person have different events and an optional attendee with an all-day event.
+    // We should see two options because each person has split the restricted times.
+    //
+    // Events  :       |--A--|     |--B--|
+    //                       |~~C~~|
+    // Day     : |-----------------------------|
+    // Options : |--1--|                 |--2--|
+
+    Collection<Event> events = Arrays.asList(
```

Comment:
One of Google's unit test best practices is to clearly separate a test into arrange/act/assert steps by only having empty lines between the steps.

Arrange = Setting up test parameters or mocks
Act = Preferrably single line that calls the method under test
Assert = Create the expected response and do your assertions / verifications

You kind of did that, but a small improvement you can do is to separate them by whitespaces so that the only empty lines would be before and after line 184 `Collection<TimeRange> actual = query.query(events, request);`, which is the act step.